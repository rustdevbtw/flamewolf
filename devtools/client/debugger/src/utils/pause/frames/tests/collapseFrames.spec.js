/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import { collapseFrames } from "../collapseFrames";

describe("collapseFrames", () => {
  it("default", () => {
    const groups = collapseFrames([
      { displayName: "a", location: { source: {} } },

      { displayName: "b", library: "React", location: { source: {} } },
      { displayName: "c", library: "React", location: { source: {} } },
    ]);

    expect(groups).toMatchSnapshot();
  });

  it("promises", () => {
    const groups = collapseFrames([
      { displayName: "a", location: { source: {} } },

      { displayName: "b", library: "React", location: { source: {} } },
      { displayName: "c", library: "React", location: { source: {} } },
      {
        displayName: "d",
        library: undefined,
        asyncCause: "promise callback",
        location: { source: {} },
      },
      { displayName: "e", library: "React", location: { source: {} } },
      { displayName: "f", library: "React", location: { source: {} } },
      {
        displayName: "g",
        library: undefined,
        asyncCause: null,
        location: { source: {} },
      },
    ]);

    expect(groups).toMatchSnapshot();
  });
});
