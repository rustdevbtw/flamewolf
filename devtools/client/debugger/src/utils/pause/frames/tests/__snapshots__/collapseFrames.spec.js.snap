// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapseFrames default 1`] = `
Array [
  Object {
    "displayName": "a",
    "location": Object {
      "source": Object {},
    },
  },
  Array [
    Object {
      "displayName": "b",
      "library": "React",
      "location": Object {
        "source": Object {},
      },
    },
    Object {
      "displayName": "c",
      "library": "React",
      "location": Object {
        "source": Object {},
      },
    },
  ],
]
`;

exports[`collapseFrames promises 1`] = `
Array [
  Object {
    "displayName": "a",
    "location": Object {
      "source": Object {},
    },
  },
  Array [
    Object {
      "displayName": "b",
      "library": "React",
      "location": Object {
        "source": Object {},
      },
    },
    Object {
      "displayName": "c",
      "library": "React",
      "location": Object {
        "source": Object {},
      },
    },
  ],
  Object {
    "asyncCause": "promise callback",
    "displayName": "d",
    "library": undefined,
    "location": Object {
      "source": Object {},
    },
  },
  Array [
    Object {
      "displayName": "e",
      "library": "React",
      "location": Object {
        "source": Object {},
      },
    },
    Object {
      "displayName": "f",
      "library": "React",
      "location": Object {
        "source": Object {},
      },
    },
  ],
  Object {
    "asyncCause": null,
    "displayName": "g",
    "library": undefined,
    "location": Object {
      "source": Object {},
    },
  },
]
`;
